lib/azure/armrest/model/base_model.rb

```diff
@@ -22,6 +22,8 @@ def self.attr_hash(*attrs)
 
       attr_hash :tags
 
+      attr_accessor :response_headers
```

Comment:
Memory will not be a concern if every model object holds a reference to the same header instance. 

I assume the header responding for a LIST request is different from the header responding for a GET request for a single object.  I therefore agree with you that it may be wrong to assign an unprocessed LIST header to each individual model object, unless the two headers are identical.

In addition, we should hide the response_header for internal use only because of the above discussed difference. Is there a use case that the user should know the header?